### ISSUE TITLE
[FEA] token_counts nvtext preprocessing method

### ISSUE THREAD

**Issue by beckernick** [rapidsai/custrings#199]
### Description

As a user, I'd like a pre-processing method that essentially gives me the `token_counts`, or the equivalent functionality so that I can create word/token vectors. `nvtext.strings_counts` returns the occurrence counts of each string pattern, not each token, because tokens can be subsets of one another.

### Example
The context is that I would like to do something like this to find the count of each unique token in the nvstrings object: 
```
s = nvstrings.to_device(["hello this is me","goodbye","", 'word'])
t = nvtext.unique_tokens(s)
n = nvtext.strings_counts(s,t)
n = np.array(n)
print(n)
print(list(zip(t.to_host(), n.sum(axis=0))))
[[0 1 2 1 1 0]
 [1 0 0 0 0 0]
 [0 0 0 0 0 0]
 [0 0 0 0 0 1]]
[('goodbye', 1), ('hello', 1), ('is', 2), ('me', 1), ('this', 1), ('word', 1)]
```

However, because strings_counts looks for the actual pattern, I end up with `is` matching twice. This is expected behavior for this method, but it's not desirable from an NLP word vectorization workflow. Adding padding wouldn't solve this problem, but that's how I noticed it.

A common NLP usage pattern is to use a vectorizer from  `sklearn.feature_extraction.text` to create word vectors and get the occurrence counts of each token in a corpus.

With sklearn, a sample usage pattern could look like this:

```
vectorizer = CountVectorizer()
X = vectorizer.fit_transform(data['text']) # create sparse matrix of word count vectors
word_index_mapping = vectorizer.vocabulary_ # dictionary of {word : index} for lookups
total_number_of_words = np.sum(X.toarray()) # just a summation to get total number of words
word_counts_mapping = dict(zip(vectorizer.get_feature_names(), X.toarray().sum(axis=0))) # associate the word with it's occurrence count
```

I can't currently seem to smoothly mimic this with custrings (abstracting from API differences), because the string occurrence counts don't correspond to the tokens. I can use nvcategory and cudf to count the hashed values, but this is onerous.
